src/store_file.rs

```diff
@@ -0,0 +1,201 @@
+  /// # Examples
+  /// ```
+  /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+  /// use tauri_plugin_store::StoreBuilder;
+  ///
+  /// let builder = StoreBuilder::new("store.bin".parse()?);
+  ///
+  /// # Ok(())
+  /// # }
+  /// ```
+  pub fn new(path: PathBuf) -> Self {
+    Self {
+      path,
+      defaults: None,
+      cache: Default::default(),
+      serialize: default_serialize,
+      deserialize: default_deserialize,
+    }
+  }
+
+  /// Inserts a default key-value pair.
+  ///   
+  /// # Examples
+  /// ```
+  /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+  /// use tauri_plugin_store::StoreBuilder;
+  /// use std::collections::HashMap;
+  ///
+  /// let mut defaults = HashMap::new();
+  ///
+  /// defaults.insert("foo".to_string(), "bar".into());
+  ///
+  /// let builder = StoreBuilder::new("store.bin".parse()?)
+  ///   .defaults(defaults);
+  ///
+  /// # Ok(())
+  /// # }
+  pub fn defaults(&mut self, defaults: HashMap<String, JsonValue>) -> &mut Self {
+    self.cache = defaults.clone();
+    self.defaults = Some(defaults);
+    self
+  }
+
+  /// Inserts multiple key-value pairs.
+  ///
+  /// # Examples
+  /// ```
+  /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+  /// use tauri_plugin_store::StoreBuilder;
+  ///
+  /// let builder = StoreBuilder::new("store.bin".parse()?)
+  ///   .default("foo".to_string(), "bar".into());
+  ///
+  /// # Ok(())
+  /// # }
+  pub fn default(&mut self, key: String, value: JsonValue) -> &mut Self {
+    self.cache.insert(key.clone(), value.clone());
+    self
+      .defaults
+      .get_or_insert(HashMap::new())
+      .insert(key, value);
+    self
+  }
+
+  /// Defines a custom serialization function.
+  ///
+  /// # Examples
+  /// ```
+  /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+  /// use tauri_plugin_store::StoreBuilder;
+  ///
+  /// let builder = StoreBuilder::new("store.json".parse()?)
+  ///   .serialize(|cache| serde_json::to_vec(&cache).map_err(Into::into));
+  ///
+  /// # Ok(())
+  /// # }
+  pub fn serialize(&mut self, serialize: SerializeFn) -> &mut Self {
+    self.serialize = serialize;
+    self
+  }
+
+  /// Defines a custom deserialization function
+  ///
+  /// # Examples
+  /// ```
+  /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+  /// use tauri_plugin_store::StoreBuilder;
+  ///
+  /// let builder = StoreBuilder::new("store.json".parse()?)
+  ///   .deserialize(|bytes| serde_json::from_slice(&bytes).map_err(Into::into));
+  ///
+  /// # Ok(())
+  /// # }
+  pub fn deserialize(&mut self, deserialize: DeserializeFn) -> &mut Self {
+    self.deserialize = deserialize;
+    self
+  }
+
+  /// Builds the [`Store`].
+  ///
+  /// # Examples
+  /// ```
+  /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+  /// use tauri_plugin_store::StoreBuilder;
+  ///
+  /// let store = StoreBuilder::new("store.bin".parse()?).build();
+  ///
+  /// # Ok(())
+  /// # }
+  pub fn build(&self) -> Store {
```

Comment:
Why `&self` here instead of `self`? Can't we consume the instance here to avoid the `.clone()`?

Reply:
Clippy was complaining previously with `self`, but it seems fine now. will fix